### PR TITLE
feat(3rd party board): Adding ViraLink IoT Gate Boards

### PR DESCRIPTION
**Pull request by armp30**
### Description of Change
feat(3rd party board): Adding ViraLink IoT Gate Boards

### Tests scenarios
I have tested my Pull Request on Arduino-esp32 core v3.0.1.

### CODE REVIEW

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
As @me-no-dev commented, please change the `build.board` to uppercase.

**Comment by me-no-dev** [comment]
@armp30 no response for over a week? Would you please finish this? If no further response, we will close it.

**Comment by armp30** [comment]
@me-no-dev fixed

File: variants/ViraLink-G0.1/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#define LED_BUILTIN LED_BUILTIN
+
+static const uint8_t RESET_KEY = 0;
+
+static const uint8_t RS485_TX = 32;
+static const uint8_t RS485_RX = 35;
+
+static const uint8_t SDA = 4;
+static const uint8_t SCL = 16;
+
```
Comment (P-R-O-C-H-Y) [inline variants/ViraLink-G0.1/pins_arduino.h:17]: Please provide default SPI pins. If you don't want to specify them, you can use -1 as shown in suggestion.
```suggestion
static const uint8_t SS = -1;
static const uint8_t MOSI = -1;
static const uint8_t SCK = -1;
static const uint8_t MISO = -1;
```

File: variants/ViraLink-G1.1/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+static const uint8_t GSM1_TX = 15;
+static const uint8_t GSM1_RX = 34;
+
+static const uint8_t GSM2_TX = 32;
+static const uint8_t GSM2_RX = 35;
+
+static const uint8_t GSM_PWR = 33;
+
+static const uint8_t SDA = 4;
+static const uint8_t SCL = 16;
```
Comment (P-R-O-C-H-Y) [inline variants/ViraLink-G1.1/pins_arduino.h:22]: Please provide default SPI pins. If you don't want to specify them, you can use -1 as shown in suggestion.
```suggestion
static const uint8_t SS = -1;
static const uint8_t MOSI = -1;
static const uint8_t SCK = -1;
static const uint8_t MISO = -1;
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@armp30 Please fix the missing default SPI pins.

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM.

**Pull request merged**